### PR TITLE
Add support for duration=None

### PR DESCRIPTION
**Pull request by Marenz**

### CODE REVIEW

File: src/frequenz/client/dispatch/types.py [hunk 1]
```diff
@@ -332,23 +338,26 @@ def to_protobuf(self) -> PBDispatch:
...
+        return PBDispatch(
+            metadata=DispatchMetadata(
+                dispatch_id=self.id,
+                create_time=to_timestamp(self.create_time),
+                modification_time=to_timestamp(self.update_time),
+            ),
+            data=DispatchData(
+                type=self.type,
+                start_time=to_timestamp(self.start_time),
+                duration=int(self.duration.total_seconds()) if self.duration else None,
```
Comment (daniel-zullo-frequenz) [inline src/frequenz/client/dispatch/types.py]: `round()` was used in `internal_types` to set the duration. You might want to use round here as well for consistency?

File: tests/test_dispatch_cli.py [hunk 2]
```diff
@@ -337,6 +337,7 @@ async def test_create_command(  # pylint: disable=too-many-arguments,too-many-lo
             ),
         )
 
+    print(result.output)
```
Comment (daniel-zullo-frequenz) [inline tests/test_dispatch_cli.py]: do you want to keep this print()?

File: tests/test_dispatch_cli.py [hunk 3]
```diff
@@ -172,9 +172,9 @@ async def test_list_command(  # pylint: disable=too-many-arguments
                 "create",
                 "829",
                 "test",
+                "BATTERY",
                 "in 1 hour",
                 "1h",
-                "BATTERY",
```
Comment (daniel-zullo-frequenz) [inline tests/test_dispatch_cli.py:177]: Is this change due to the positional arguments change the order as now there might not be a duration? I just want to be sure I got this correctly

**Review by daniel-zullo-frequenz — COMMENTED** [review]
I have a few questions to check for. LGTM otherwise

File: tests/test_dispatch_cli.py (hunk 3 above)
Comment (Marenz) [reply tests/test_dispatch_cli.py:177]: Yep, this change is due to the fact that it can now be skipped, so I moved it to the end (as those parameters are position based, not named based)

**Comment by Marenz** [comment]
updated

**Review by daniel-zullo-frequenz — DISMISSED** [review]

File: src/frequenz/client/dispatch/_internal_types.py [hunk 4]
```diff
@@ -25,6 +25,8 @@
     component_selector_to_protobuf,
 )
 
+# pylint: enable=no-name-in-module
+
```
Comment (llucax) [inline src/frequenz/client/dispatch/_internal_types.py:29]: This doesn't look right here, but you could also just disable it globally in the whole project as `mypy` will check for this anyway.

File: src/frequenz/client/dispatch/_internal_types.py [hunk 5]
```diff
@@ -42,8 +44,11 @@ class DispatchCreateRequest:
     start_time: datetime
     """The start time of the dispatch in UTC."""
 
-    duration: timedelta
-    """The duration of the dispatch, represented as a timedelta."""
+    duration: timedelta | None
+    """The duration of the dispatch, represented as a timedelta.
+
+    If None, the dispatch is considered to be "infinite" or "no-duration".
```
Comment (llucax) [inline src/frequenz/client/dispatch/_internal_types.py]: For me "no-duration" is not super clear, I think I would say "instantaneous" (or maybe both), or use an example as you did in the docs above.

File: src/frequenz/client/dispatch/__main__.py [hunk 6]
```diff
@@ -239,9 +239,9 @@ def validate_reccurance(ctx: click.Context, param: click.Parameter, value: Any)
     required=True,
     type=str,
 )
-@click.argument("start-time", required=True, type=FuzzyDateTime())
-@click.argument("duration", required=True, type=FuzzyTimeDelta())
 @click.argument("selector", required=True, type=SelectorParamType())
+@click.argument("start-time", required=True, type=FuzzyDateTime())
+@click.argument("duration", required=False, type=FuzzyTimeDelta())
```
Comment (llucax) [inline src/frequenz/client/dispatch/__main__.py:244]: In the second commit there are quite a few reordering changes that don't seem related, here and also in many imports or string lists.

**Review by llucax — COMMENTED** [review]
LGTM besides the small optional comments. The only thing that I would probably add is a test for receiving a dispatch with no duration, not sure if it is tested indirectly in the CLI test.

File: src/frequenz/client/dispatch/_internal_types.py (hunk 4 above)
Comment (Marenz) [reply src/frequenz/client/dispatch/_internal_types.py:29]: I think this is mostly at this place due to `black`

File: src/frequenz/client/dispatch/__main__.py (hunk 6 above)
Comment (Marenz) [reply src/frequenz/client/dispatch/__main__.py:244]: Yeah, I blame black.

Comment (Marenz) [reply src/frequenz/client/dispatch/__main__.py:244]: Ah no, the reordering here is very much related. `duration` became an optional positional parameter, so it had to move to the end

File: src/frequenz/client/dispatch/_internal_types.py (hunk 4 above)
Comment (llucax) [reply src/frequenz/client/dispatch/_internal_types.py:29]: Yeah, it is weird, it has to be just before the next line.

File: src/frequenz/client/dispatch/__main__.py (hunk 6 above)
Comment (llucax) [reply src/frequenz/client/dispatch/__main__.py:244]: Really, if black is reordering that it should be a nasty bug, decorators are not commutable, if you change the order the results could be completely different!

**Review by daniel-zullo-frequenz — APPROVED** [review]
